spec/models/merchant_spec.rb

```diff
@@ -33,6 +34,16 @@
         expect(@merchant1.unique_invoices).to match([@invoice1, @invoice2])
       end
     end
+
+    describe '#invoice_items_for_this_invoice' do
+      it 'returns invoice items only for this invoice' do
+        expect(@merchant1.items_for_this_invoice(@invoice1)).to match([@invoiceitem1, @invoiceitem2])
+      end
+    end
+
+    describe '#invoice_revenue' do
+      it 'returns the total revenue for items sold on this invoice' do
+        expect(@merchant1.invoice_revenue(@invoice1)).to eq(5400)
```

Comment:
*Needs 2 'ends' here to close off the #invoice revenue test
Lines 40 ad 46 should have @invoice1.id as the argument instead of @INVOICE1 object for clarity (the test passes without the change, however for developer empathy should include id.)